GitHub contributing (#1317)
nasa-gcn/gcn.nasa.gov: app/routes/docs.contributing.github/route.mdx

---
handle:
  breadcrumb: GitHub
---

import copyUrl from './copy-url.png'
import githubFork from './github-fork.png'
import githubImage from './github.png'
import newFork from './new-fork.png'

# GitHub

> We maintain several repositories under the [nasa-gcn](https://github.com/nasa-gcn) GitHub organization. These include kafka wrappers for [python](https://github.com/nasa-gcn/gcn-kafka-python) and [JavaScript](https://github.com/nasa-gcn/gcn-kafka-js), the new [email distribution service](https://github.com/nasa-gcn/gcn-email), and several other projects that support the new GCN.

Reviewer: I think a list would work better here:

```suggestion
GCN consists of several open-source projects grouped into the [nasa-gcn](https://github.com/nasa-gcn) GitHub organization. These include:

* [nasa-gcn/gcn.nasa.gov](https://github.com/nasa-gcn/gcn.nasa.gov): This web site
* [nasa-gcn/gcn-schema](https://github.com/nasa-gcn/gcn.nasa.gov): The GCN Notices unified schema
* [nasa-gcn/gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python), [nasa-gcn/gcn-kafka-js](https://github.com/nasa-gcn/gcn-kafka-js): Kafka client libraries
...
```

---
handle:
  breadcrumb: GitHub
---

import copyUrl from './copy-url.png'
import githubFork from './github-fork.png'
import githubImage from './github.png'
import newFork from './new-fork.png'

# GitHub

GCN consists of several open-source projects grouped into the [nasa-gcn](https://github.com/nasa-gcn) GitHub organization. These include:

- [nasa-gcn/gcn.nasa.gov](https://github.com/nasa-gcn/gcn.nasa.gov): This web site
- [nasa-gcn/gcn-schema](https://github.com/nasa-gcn/gcn.nasa.gov): The GCN Notices unified schema
- [nasa-gcn/gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python), [nasa-gcn/gcn-kafka-js](https://github.com/nasa-gcn/gcn-kafka-js): Kafka client libraries
  ...

To contribute to any project in the [nasa-gcn](https://github.com/nasa-gcn) GitHub organization, follow these steps:

- [Fork a copy of our repo](#forking-a-repo)

- [Clone your repo to your machine](#cloning-a-repo)

- [Create a branch for your new changes](#branching-off-main)

> - [Create a Pull Request to contribute to the original project](#pull-requests)

Reviewer: - [ ] Avoid unnecessary abbreviations: replace "repo" with "repository."
- [ ] Can the items in this list match the titles of the sections verbatim?

> You will need to install Git on your development machine. Check out [the official installation instructions](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).

Reviewer: This should come _before_ all of the above steps.

## Forking a Repo

> Navigate to the page repository page for whichever project you wish to contribute to, for example: [gcn.nasa.gov](https://github.com/nasa-gcn/gcn.nasa.gov).

Reviewer: Typo: "page repository page"

<img
  src={githubImage}
  width="712"
  height="392"
  className="height-auto"
  alt="gcn.nasa.gov GitHub page"
/>

Click the Fork button.

<img
  src={githubFork}
  width="712"
  height="392"
  className="height-auto"
  alt="Highlighted fork button"
/>

Click Create Fork to create a new repo in your account.

<img
  src={newFork}
  width="712"
  height="392"
  className="height-auto"
  alt="Highlighted new fork"
/>

## Cloning a Repo

> We recommend checking out [this guide](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account) on adding an SSH key to your github account.

Reviewer: Can we put this, along with the Git installation instructions, in a "Setup" or "Prerequisites" section?

With your new fork now created, click the Code button and copy the URL in the SSH tab.

<img
  src={copyUrl}
  width="712"
  height="392"
  className="height-auto"
  alt="Highlighted URL"
/>

Open a terminal window and run the following

```sh
git clone <your copied repo url>
```

> ## Branching off main

Reviewer: ```suggestion
## Creating a branch
```

In a terminal, set your current working directory to the cloned directory and run the following to create and switch to a new branch for your code changes:

```sh
git checkout -b <your branch name>
```

To stage your changes:

```sh
git add .
```

Create a new local commit and add a descriptive comment:

```sh
git commit -m "Your descriptive comment"
```

Push your commits out to your remote fork:

```sh
git push
```

## Pull Requests

Once you have completed your changes on your branch, create a [pull request](https://github.com/nasa-gcn/gcn.nasa.gov/pulls).

To create a pull request, navigate to your fork in your own github repo. Click the "branches" link in the upper left hand secondary nav menu. Once you are on the branches index, click the "New pull request" button on the right hand side of the branch you would like to use. Once you have filled in your pull request title and description, click the "Create pull request" button on the lower left hand side to create the pull request.

## Issues

> We use the GitHub issues pages to track bugs and other problems. To submit a new issue, navigate to the appropriate [nasa-gcn](https://github.com/nasa-gcn/) repository and click on Issues.

Reviewer: ```suggestion
We use the GitHub issues to track bugs and feature requests. To submit a new issue, navigate to the appropriate [nasa-gcn](https://github.com/nasa-gcn/) repository and click on Issues.
```

> Click New Issue and fill out the form. Please include a concise and descriptive title and as much relevant information in the comment as possible. This includes steps to reproduce issues, screenshots, error messages, and anything else that may be helpful for us or others to pinpoint the cause of an issue.

Reviewer: ```suggestion
Click New Issue and fill out the form. Please include a concise and descriptive title and as much relevant information in the comment as possible. If relevant, include steps to reproduce, screenshots, error messages, and anything else that may be helpful for us or others to pinpoint the cause of an issue.
```
Please add a screen shot or two for issues.

---
handle:
  breadcrumb: GitHub
---

import copyUrl from './copy-url.png'
import githubFork from './github-fork.png'
import githubImage from './github.png'
import newFork from './new-fork.png'

# GitHub

GCN consists of several open-source projects grouped into the [nasa-gcn](https://github.com/nasa-gcn) GitHub organization. These include:

- [nasa-gcn/gcn.nasa.gov](https://github.com/nasa-gcn/gcn.nasa.gov): This web site
- [nasa-gcn/gcn-schema](https://github.com/nasa-gcn/gcn.nasa.gov): The GCN Notices unified schema
- [nasa-gcn/gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python), [nasa-gcn/gcn-kafka-js](https://github.com/nasa-gcn/gcn-kafka-js): Kafka client libraries
  ...

To contribute to any project in the [nasa-gcn](https://github.com/nasa-gcn) GitHub organization, follow these steps:

> - [Forking a Repository](#forking-a-repository)

Reviewer: Add a bullet point and link for setting up Git.

- [nasa-gcn/gcn-kafka-python](https://github.com/nasa-gcn/gcn-kafka-python), [nasa-gcn/gcn-kafka-js](https://github.com/nasa-gcn/gcn-kafka-js): Kafka client libraries
  ...

To contribute to any project in the [nasa-gcn](https://github.com/nasa-gcn) GitHub organization, follow these steps:

- [Forking a Repository](#forking-a-repository)

- [Cloning a Repository](#cloning-a-repository)

- [Creating a Branch](#creating-a-branch)

- [Pull Requests](#pull-requests)

> 

Reviewer: - [ ] Switch the order of these two sections.
- [ ] Make the prerequisites section a second-level heading (`##`).
- [ ] Make sure that the headings are all the same kinds of phrase (gerund phrases are fine). So for example, rename "Prerequisites" to something like "Setting up Git on your computer"

<img
  src={githubFork}
  width="712"
  height="392"
  className="height-auto"
  alt="Highlighted fork button"
/>

Click Create Fork to create a new repository in your account.

<img
  src={newFork}
  width="712"
  height="392"
  className="height-auto"
  alt="Highlighted new fork"
/>

## Cloning a Repository

With your new fork now created, click the Code button and copy the URL in the SSH tab.

<img
  src={copyUrl}
  width="712"
  height="392"
  className="height-auto"
  alt="Highlighted URL"
/>

Open a terminal window and run the following

```sh
git clone <your copied repo url>
```

## Creating a Branch

In a terminal, set your current working directory to the cloned directory and run the following to create and switch to a new branch for your code changes:

```sh
git checkout -b <your branch name>
```

To stage your changes:

```sh
git add .
```

Create a new local commit and add a descriptive comment:

```sh
git commit -m "Your descriptive comment"
```

Push your commits out to your remote fork:

```sh
git push
```

## Pull Requests

Once you have completed your changes on your branch, create a [pull request](https://github.com/nasa-gcn/gcn.nasa.gov/pulls).

> To create a pull request, navigate to your fork in your own github repository. Click the "branches" link in the upper left hand secondary nav menu. Once you are on the branches index, click the "New pull request" button on the right hand side of the branch you would like to use. Once you have filled in your pull request title and description, click the "Create pull request" button on the lower left hand side to create the pull request.

Reviewer: ```suggestion
To create a pull request, navigate to your fork in your own github repository. Click the "Branches" link in the upper left hand secondary nav menu. On the next page, click the "New pull request" button for the branch you would like to use. Fill in the title and description, then click the "Create pull request" button on the lower left hand side to create the pull request.
```
